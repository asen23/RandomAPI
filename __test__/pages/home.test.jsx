import { render, screen } from "@testing-library/react"
import Home from "pages/index"

describe("Home", () => {
    it("renders unchanged", () => {
        const { asFragment } = render(<Home />)
        expect(asFragment()).toMatchSnapshot()
    })

    it("renders two heading", () => {
        render(<Home />)

        const firstHeading = screen.getByRole("heading", {
            name: "Welcome to random API",
        })
        expect(firstHeading).toBeInTheDocument()

        const secondHeading = screen.getByRole("heading", {
            name: "This website provide random API interfaces",
        })
        expect(secondHeading).toBeInTheDocument()
    })
})
