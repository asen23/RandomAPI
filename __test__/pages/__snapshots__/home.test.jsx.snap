// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders unchanged 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex justify-center items-center flex-col min-h-screen gap-8 bg-primary"
    >
      <h1
        class="text-3xl font-bold"
      >
        Welcome to random API
      </h1>
      <h2
        class="text-2xl font-bold"
      >
        This website provide random API interfaces
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
